app/chatpal-search/server/provider/provider.js

```diff
@@ -313,6 +313,7 @@ class ChatpalProvider extends SearchProvider {
 		if (!this.index) { return callback({ msg:'Chatpal_currently_not_active' }); }
 
 		const type = payload.resultType === 'All' ? ['message', 'user', 'room'] : ['message'];
+		const params = Object.assign({}, payload.custom);
```

Comment:
where/how can `payload.custom` be set? wondering if something harmful can be sent as there are no validation here.

Reply:
Hi Diego! The custom payload can be set by everyone who uses the meteor search function. This feature can be helpful for some extensions of chatpal. It is validated by the Chatpal Server API, so no problem!